cognite/neat/workflows/steps/lib/graph_loader.py

```diff
@@ -218,48 +231,63 @@ def run(  # type: ignore[override]
-
-            orphan_assets, circular_assets = validate_asset_hierarchy(rdf_asset_dicts)
-
-            logging.info(orphan_assets)
+            if asset_cleanup_type in ["orphans", "full"]:
+                logging.info("Removing orphaned assets")
+                for external_id in orphan_assets:
+                    del rdf_asset_dicts[external_id]
+            else:
+                orphanage_asset_external_id = (
+                    f"{rules.rules.metadata.externalIdPrefix}orphanage-{rules.dataset_id}"
+                    if rules.rules.metadata.externalIdPrefix
+                    else "orphanage"
+                )
+
+                # Kill the process if you dont have orphanage asset in your asset hierarchy
+                # and inform the user that it is missing !
+                if orphanage_asset_external_id not in rdf_asset_dicts:
+                    msg = f"You dont have Orphanage asset {orphanage_asset_external_id} in asset hierarchy!"
+                    logging.error(msg)
+                    raise Exception(msg)
+
+                logging.error("Orphaned assets will be assigned to 'Orphanage' root asset")
+
+                for external_id in orphan_assets:
+                    rdf_asset_dicts[external_id]["parent_external_id"] = orphanage_asset_external_id
         else:
             logging.info("No orphaned assets found, your assets look healthy !")
 
         if circular_assets:
-            msg = f"Found circular dependencies: {circular_assets!s}"
-            logging.error(msg)
-            raise Exception(msg)
-        elif orphan_assets:
-            msg = f"Not able to fix orphans: {', '.join(orphan_assets)}"
-            logging.error(msg)
-            raise Exception(msg)
+            logging.error(f"Found circular dependencies: {circular_assets}")
+            if asset_cleanup_type in ["circular", "full"]:
+                logging.info("Removing circular assets")
+                for circular_path in circular_assets:
+                    circular_external_id = circular_path[-1]
+                    del rdf_asset_dicts[circular_external_id]
         else:
             logging.info("No circular dependency among assets found, your assets hierarchy look healthy !")
 
+        # if full cleanup is selected, we dont need to check for circular and orphan assets [OPTIMIZATION]
+        if asset_cleanup_type != "full":
+            orphan_assets, circular_assets = validate_asset_hierarchy(rdf_asset_dicts)
+            if circular_assets:
+                msg = f"Found circular dependencies: {circular_assets!s}"
+                logging.error(msg)
+                raise Exception(msg)
+            elif orphan_assets:
+                msg = f"Not able to fix orphans: {', '.join(orphan_assets)}"
+                logging.error(msg)
+                raise Exception(msg)
+            else:
+                logging.info("No circular dependency among assets found, your assets hierarchy look healthy !")
+
```

Comment:
I am wandering would second pass of `validate_asset_hierarchy` be useful to capture any potentially newly created orphans/circular dependency?